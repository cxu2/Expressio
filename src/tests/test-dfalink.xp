main : -> int {
dfa help;
help  = {
 states : 7
 alphabet : ['0', '1', '2', '3', '4', '5']
 start : 4
 final : [5, 2]
 transitions : [ (0 '0' 4), (1 '0' 3) ] 
};


link(help, 4, '2', 5);

printdfa(help);

  return 0;
}
