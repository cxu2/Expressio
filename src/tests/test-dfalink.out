
nstates:  7
alphabet: 0 1 2 3 4 5 
nsym:     6
start:    4
fin:      5 2 
nfin:     2
delta:    
        |   0 |   1 |   2 |   3 |   4 |   5 
    ----|-----|-----|-----|-----|-----|-----
       0|   4    -1    -1    -1    -1    -1  
       1|   3    -1    -1    -1    -1    -1  
       2|  -1    -1    -1    -1    -1    -1  
       3|  -1    -1    -1    -1    -1    -1  
       4|  -1    -1     5    -1    -1    -1  
       5|  -1    -1    -1    -1    -1    -1  
       6|  -1    -1    -1    -1    -1    -1  
